hprotoc/Text/ProtocolBuffers/ProtoCompile/Resolve.hs

```diff
@@ -1370,7 +1370,7 @@ loadProto' fdpReader protoFile = goState (load Set.empty protoFile) where
             return global'env
 
 loadFailed :: (Monad m) => LocalFP -> String -> m a
-loadFailed f msg = fail . unlines $ ["Parsing proto:",show (unLocalFP f),"has failed with message",msg]
```

Comment:
This doesn't seem to be right. I think `loadFailed` should have a `MonadFail` constraint instead.

Reply:
It is impossible, because below, in `loadCodeGenRequest`, `loadFailed` is used with `Identity` monad, which does not have `MonadFail` instance.